packages/devextreme/js/__internal/ui/chat/messagebox.ts

```diff
@@ -68,27 +83,57 @@ class MessageBox extends DOMComponent<MessageBox, Properties> {
     this._createMessageEnteredAction();
     this._createTypingStartAction();
     this._createTypingEndAction();
+    this._createMessageEditCanceledAction();
   }
 
   _initMarkup(): void {
     $(this.element()).addClass(CHAT_MESSAGEBOX_CLASS);
 
     super._initMarkup();
 
-    this._renderTextArea();
-    this._renderButton();
+    if (this.option('editingText')) {
+      this._renderEditingPreview();
+    }
+
+    this._renderInputContainer();
   }
 
-  _renderTextArea(): void {
+  _renderInputContainer(): void {
+    const $messageBox = $('<div>')
+      .addClass(CHAT_MESSAGEBOX_INPUT_CONTAINER_CLASS)
+      .appendTo(this.element());
+
+    const $textArea = this._renderTextArea();
+    const sendButton = this._renderButton();
+
+    $messageBox
+      .append($textArea)
+      .append(sendButton);
+  }
+
+  _cancelMessageEdit(): void {
+    this.option('editingText', undefined);
+    this._messageEditCanceledAction?.();
```

Comment:
You don't use arguments that action provides, so the common callback will sufficient here. Let's simplify it